### ISSUE TITLE
.offset() adds scrollTop to the top coordinate (only in IE and jQuery 1.12.0 and 2.2.0)

### ISSUE THREAD

**Issue by benjamin-albert** [jquery/jquery#2897]
## Context
I'm a collaborator in the [jQuery UI Month Picker](https://kidsysco.github.io/jquery-ui-month-picker/) project.

Recently we've setup a demo page where we've used the latest compat version of jQuery (1.12.0).

The plugin allows the user to click a button that uses the jQuey UI [.position()](https://jqueryui.com/position/) method to position a menu under an input field.

As [demonstrated in this page] 
(https://benjamin-albert.github.io/jquery-ui-month-picker/jquery-bug.html), everything worked fine except for **Internet Explorer** which seems to positions the menu as far below the input as you scrolled.

**IMPORTANT**: This only happens the **first time** after you position an element, but you can reproduce the issue by refreshing the page.

## What have I tried
At first I thought this had something to do with the jQuery UI version (1.11.4) we were using, but switching to several different versions of jQuery UI didn't solve the issue.

Then I tried to use jQuery version 1.11.3 and everything is working fine even in IE.

I than tried version 1.9.1 (which we run [our tests](https://github.com/KidSysco/jquery-ui-month-picker#month-picker-development-kit) against) and once again everything worked find.

I also tried a different version 1.9.2 of jQuery UI with jQuery 1.9.1 and everything worked fine.

Then I tried to use jQuery version 2.2.0 and one again everyting was working fine except for IE.

## What's going on in the code
When the user clicks the button the [Open() method](https://github.com/KidSysco/jquery-ui-month-picker/blob/c4ce4d6ae9cf2d7a079e809e627c1bf9af40d1bc/MonthPicker.js#L567) in our code is called.

The open method calls the [.fadeToggle()](http://api.jquery.com/fadetoggle/#fadeToggle-options) method using the `start` callback option.

The start callback is called right after the `dispaly:none` is removed and right before the animation begins (which as mentioned in the [jQuery ui .position() docs](http://api.jqueryui.com/position/) is **required** for it to work).

[Our start callback](https://github.com/KidSysco/jquery-ui-month-picker/blob/c4ce4d6ae9cf2d7a079e809e627c1bf9af40d1bc/MonthPicker.js#L775) calls jQuery UI's .position() method, which [eventually will call elem.offset({...})](https://github.com/jquery/jquery-ui/blob/cb69f0025f257c28ac3be39c4269373a9fa74002/ui/position.js#L336).

Eventually the [jQuery.offset.setOffset()](https://github.com/jquery/jquery/blob/29370190605ed5ddf5d0371c6ad886a4a4b5e0f9/src/offset.js#L35) method calls `elem.offset()` to get the elements offset.

Eventually the `elem.offset()` method calls [rect = elem.getBoundingClientRect();](https://github.com/jquery/jquery/blob/29370190605ed5ddf5d0371c6ad886a4a4b5e0f9/src/offset.js#L101) which seems to add `scrollTop` to the top property return by `.getBoundingClientRect();`.

## TLDR
I'm not sure if the problem is a result of us doing something wrong which just happens to work in every different configuration we've tested, or is this an issue with jQuery that needs to be resolved.

**Comment by mgol** [comment]
Looks like a duplicate of #2828 & #2836; let's follow in the former.
